packages/target-tools/src/telemetryProvider.spec.js

```diff
@@ -111,56 +121,105 @@ describe("TelemetryProvider", () => {
       })
     );
   });
-
-  it("assigns local execution mode", () => {
+  it("assigns edge execution mode for server side", () => {
     const provider = TelemetryProvider();
 
+    const DELIVERY_RESPONSE_SERVER_SIDE = {
+      status: 200,
+      requestId: "7a568cbfe3f44f0b99d1092c246660c1",
+      client: "targettesting",
+      id: {
+        tntId: "338e3c1e51f7416a8e1ccba4f81acea0.27_0",
+        marketingCloudVisitorId: "07327024324407615852294135870030620007"
+      },
+      edgeHost: "mboxedge28.tt.omtrdc.net",
+      telemetryServerToken: "encryptedtelemetry"
+    };
+
     provider.addDeliveryRequestEntry(
       TARGET_REQUEST,
       TARGET_TELEMETRY_ENTRY,
-      STATUS_OK,
-      DECISIONING_METHOD.ON_DEVICE
+      DELIVERY_RESPONSE_SERVER_SIDE,
+      DECISIONING_METHOD.SERVER_SIDE
     );
+    const entries = provider.getAndClearEntries();
+    expect(entries[0].mode).toEqual(EXECUTION_MODE.EDGE);
+  });
+
+  it("assigns local execution mode for on device", () => {
+    const provider = TelemetryProvider();
+    const DELIVERY_RESPONSE_ON_DEVICE = {
+      status: 206,
+      requestId: "7a568cbfe3f44f0b99d1092c246660c2",
+      client: "targettesting",
+      id: {
+        tntId: "338e3c1e51f7416a8e1ccba4f81acea0.28_0",
+        marketingCloudVisitorId: "07327024324407615852294135870030620007"
+      },
+      edgeHost: null,
+      telemetryServerToken: "encryptedtelemetry"
+    };
+
     provider.addDeliveryRequestEntry(
       TARGET_REQUEST,
       TARGET_TELEMETRY_ENTRY,
-      STATUS_OK,
-      DECISIONING_METHOD.HYBRID
+      DELIVERY_RESPONSE_ON_DEVICE,
+      DECISIONING_METHOD.SERVER_SIDE
```

Comment:
I think this should be DECISIONING_METHOD.ON_DEVICE

Reply:
oho, I realized the mess up.... Thanks for  catching it.